tests/integration/test_bundle.py

```diff
@@ -23,15 +37,41 @@ def get_this_script_dir() -> Path:
     return Path(path)
 
 
+class State:
```

Comment:
Nitpick:

I am a bit concerned about this class. From its name, it seems like it can encapsulate anything related to the test states. So people might add more attributes in the future whenever they want to pass some other states between tests. This leads to a possible risk of misuse and changes in the attributes of the object.

I can understand what this class is for passing the knowledge of some variables from one test to another test. In another way, this also shows that we have dependencies in some of our test cases. IMHO, for testing we should make sure each test case is independent as much as possible.

I think the dependency existing between `test_create_hydra_client`, `test_authorization_code_flow`, and `test_client_credentials_flow` is better to be removed.

Reply:
While in principle I agree with you, I don't think that in integration tests keeping each test case isolated is that important. In integration tests you may want to test a flow and you may use functions to separate the different steps of a flow (else a test/function would get too big). For example, I was thinking of splitting `test_authorization_code_flow` even further.

I will remove the State and simply create a new client on each function, but let's keep in mind that all of these test cases run on a cluster that was created on earlier steps of the tests and that was altered (e.g. scaled) on other test cases.

Reply:
ic~ thanks for the clarifications.